components/login.js

```
@@ -42,21 +42,34 @@ const authErrorMessages = {
   OAuthCallback: 'Error handling OAuth response. Try again or choose a different method.',
   OAuthCreateAccount: 'Could not create OAuth account. Try again or choose a different method.',
   EmailCreateAccount: 'Could not create Email account. Try again or choose a different method.',
-  Callback: 'Try again or choose a different method.',
+  Callback: 'Could not authenticate. Try again or choose a different method.',
   OAuthAccountNotLinked: 'This auth method is linked to another account. To link to this account first unlink the other account.',
   EmailSignin: 'Failed to send email. Make sure you entered your email address correctly.',
-  CredentialsSignin: 'Auth failed. Try again or choose a different method.',
+  CredentialsSignin: 'Could not authenticate. Try again or choose a different method.',
   default: 'Auth failed. Try again or choose a different method.'
 }
 
 export function authErrorMessage (error) {
   return error && (authErrorMessages[error] ?? authErrorMessages.default)
 }
 
-export default function Login ({ providers, callbackUrl, multiAuth, error, text, Header, Footer }) {
+export default function Login ({ providers, callbackUrl, multiAuth, error, text, Header, Footer, signin }) {
   const [errorMessage, setErrorMessage] = useState(authErrorMessage(error))
   const router = useRouter()
 
+  // signup/signin awareness cookie
+  useEffect(() => {
+    const cookieOptions = [
+      `signin=${!!signin}`,
+      'path=/',
+      'max-age=' + 60 * 60 * 24, // 24 hours
+      'SameSite=Lax',
+      process.env.NODE_ENV === 'production' ? 'Secure' : ''
+    ].filter(Boolean).join(';')
+
+    document.cookie = cookieOptions
+  }, [signin])
```

Won't this prevent account creation for 24 hours if they visit the `login` page before the `sign up` page?

---

It gets overwritten by `signin=false` in that case because `signup.js` don't have the `signin` prop

---

Oh, but we stopped checking for the value on the server, only that the cookie exists, so I think @huumn is right.

I think it's better to remove this cookie in a `useEffect` cleanup function.

I also wanted to test this cookie with account switching more but during my testing yesterday, it seemed to worked as expected except that it just silently fails if the account we want to add doesn't exist but that's related to error handling and we had no error handling for account switching before, too.

---

Ah yes, you're right mb I thought of that case previously with the check for value, we can go with the useEffect. Pushing it in a bit ^^